### PR TITLE
feat(NUM-1874) Adds new user notifications

### PR DESCRIPTION
**Pull request by danielatomoiaga**

### CODE REVIEW

File: src/main/java/de/vitagroup/num/service/UserService.java [hunk 1]
```diff
@@ -288,18 +295,57 @@ private void addRoles(User user) {
     return users;
   }
 
-  private List<Notification> collectNotification(String userId) {
+  private List<Notification> collectProfileUpdateNotification(
+      String userId, String loggedInUserId) {
```
Comment (alexkarle) [inline src/main/java/de/vitagroup/num/service/UserService.java:299]: Isn't that only about user name changes? If yes shouldn't this be renamed to collectUsernameUpdate....?

File: src/main/java/de/vitagroup/num/service/notification/dto/account/ProfileUpdateNotification.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package de.vitagroup.num.service.notification.dto.account;
+
+import de.vitagroup.num.service.email.MessageSourceWrapper;
+import de.vitagroup.num.service.notification.dto.Notification;
+import java.time.Year;
+import lombok.Builder;
+
+public class ProfileUpdateNotification extends Notification {
```
Comment (alexkarle) [inline src/main/java/de/vitagroup/num/service/notification/dto/account/ProfileUpdateNotification.java:8]: Should this be renamed to UsernameUpdateNot....?

File: src/main/java/de/vitagroup/num/domain/Roles.java [hunk 3]
```diff
@@ -51,4 +51,12 @@ public static boolean isAllowedToSet(String roleToSet, List<String> callersRoles
   public static boolean hasRole(String role, Jwt principal) {
     return extractRoles(principal).contains(role);
   }
+
+  public static boolean isSuperAdmin(List<String> roles) {
```
Comment (alexkarle) [inline src/main/java/de/vitagroup/num/domain/Roles.java:55]: If that was not checked before, isn't that a bugfix then? Would be good to comment it in the story.

**Review by alexkarle — CHANGES_REQUESTED** [review]

File: src/main/java/de/vitagroup/num/domain/Roles.java (hunk 3 above)
Comment (danielatomoiaga) [reply src/main/java/de/vitagroup/num/domain/Roles.java:55]: This is just a convenience method as I restructured some code, no bug here.

**Review by alexkarle — APPROVED** [review]

**Pull request merged**